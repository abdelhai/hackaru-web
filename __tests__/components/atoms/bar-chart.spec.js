import Factory from '@/__tests__/__setups__/factory';
import BarChart from '@/components/atoms/bar-chart';

describe('BarChart', () => {
  let factory;

  beforeEach(() => {
    factory = new Factory(BarChart, {
      propsData: {
        chartData: {
          labels: [
            'Jan',
            'Feb',
            'Mar',
            'Apr',
            'May',
            'Jun',
            'Jul',
            'Aug',
            'Sep',
            'Oct',
            'Nov',
            'Dec'
          ],
          datasets: [
            {
              label: 'Development',
              backgroundColor: '#ff0',
              data: [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10]
            },
            {
              label: 'Review',
              backgroundColor: '#ff0',
              data: [10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10, 10]
            }
          ]
        }
      }
    });
  });

  it('render correctly', () => {
    const wrapper = factory.shallow();
    expect(wrapper.element).toMatchSnapshot();
  });

  it('show tooltip text correctly', () => {
    const wrapper = factory.shallow();
    expect(wrapper.vm.options.tooltips.callbacks.label({ yLabel: 10 })).toBe(
      '00:10'
    );
  });
});
