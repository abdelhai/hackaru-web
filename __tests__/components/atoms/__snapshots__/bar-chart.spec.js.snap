// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BarChart render correctly 1`] = `
<div
  class=""
>
  <div
    class="chartjs-render-monitor"
    height="400"
    id="bar-chart"
    style="display: block;"
    width="400"
  />
</div>
`;
